source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -921,13 +913,39 @@ namespace
         doc->removeChild(uniqueName);
     }
 
-    void writeNodeGraph(const PvtPrim * src, DocumentPtr dest)
+    void writeNodeGraph(const PvtPrim* src, DocumentPtr dest)
     {
         NodeGraphPtr destNodeGraph = dest->addNodeGraph(src->getName());
         writeMetadata(src, destNodeGraph, nodegraphMetadata);
 
         RtNodeGraph nodegraph(src->hnd());
 
+        // Write inputs/parameters.
+        RtObjTypePredicate<RtInput> inputsFilter;
+        for (RtAttribute attr : src->getAttributes(inputsFilter))
+        {
+            RtInput nodegraphInput = nodegraph.getInput(attr.getName());
+            if (nodegraphInput.isUniform())
+            {
+                destNodeGraph->addParameter(nodegraphInput.getName(), nodegraphInput.getType());
+            }
+            else
+            {
+                InputPtr input = destNodeGraph->addInput(nodegraphInput.getName(), nodegraphInput.getType());
```

Comment:
Hey @niklasharrysson , we can't support connections outside of the runtime so none of these graphs will be understandable will they? Also, this adds connections but doesn't really do anything for interface name publishing which what I thought we would handle first. i.e. connections would be stored as interfacename references and these references would create new connections on load.

Reply:
My thinking is that our export can support this. Since we will not allow it for now in the runtime content creation (commands for creating connections will not allow it), so no content will be exported with external connections. But once we do support it in content creation our exporter will already support it.

For interfacename publishing this is already done when writing the nodes below.

Reply:
I missed this translation. I think we'll need to allow interfacenames w/o nodedef names to avoid validation problem but can look into this afterwards. Thanks.